tests/saitests/switch.py

```diff
@@ -813,6 +813,24 @@ def sai_thrift_read_headroom_pool_watermark(client, buffer_pool_id):
         return None
     return wm_vals[0]
 
+def sai_thrift_read_queue_occupancy(client, port_id):
+    queue_list=[]
+    port_attr_list = client.sai_thrift_get_port_attribute(port_list[port_id])
+    attr_list = port_attr_list.attr_list
+    for attribute in attr_list:
+        if attribute.id == SAI_PORT_ATTR_QOS_QUEUE_LIST:
+            for queue_id in attribute.value.objlist.object_id_list:
+                queue_list.append(queue_id)
+    cnt_ids=[SAI_QUEUE_STAT_CURR_OCCUPANCY_BYTES]
```

Comment:
So looks like it uses the current queue occupy stats? if so, how it works as watermark?

Reply:
The `sai_thrift_read_queue_occupancy` function here is only used for the `fill_leakout_plus_one` utility to detect a change in queue occupancy. It is not directly used for the test except in the leakout accounting utility. The test still calls `sai_thrift_read_buffer_pool_watermark` to check the watermark.